src/SprykerEco/Zed/Braintree/Persistence/BraintreeRepository.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Braintree\Persistence;
+
+use Generated\Shared\Transfer\PaymentBraintreeTransactionRequestLogTransfer;
+use Generated\Shared\Transfer\PaymentBraintreeTransactionStatusLogTransfer;
+use Generated\Shared\Transfer\PaymentBraintreeTransfer;
+use Orm\Zed\Braintree\Persistence\Map\SpyPaymentBraintreeTransactionRequestLogTableMap;
+use Orm\Zed\Braintree\Persistence\Map\SpyPaymentBraintreeTransactionStatusLogTableMap;
+use Propel\Runtime\ActiveQuery\Criteria;
+use Propel\Runtime\Propel;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+
+/**
+ * @method \SprykerEco\Zed\Braintree\Persistence\BraintreePersistenceFactory getFactory()
+ */
+class BraintreeRepository extends AbstractRepository implements BraintreeRepositoryInterface
+{
+    /**
+     * @param int $idPaymentBraintree
+     *
+     * @return \Generated\Shared\Transfer\PaymentBraintreeTransfer|null
+     */
+    public function findPaymentBraintreeById(int $idPaymentBraintree): ?PaymentBraintreeTransfer
+    {
+        $paymentBraintree = $this->getFactory()
```

Comment:
Please rename variables in this class that might have entities according to our naming conventions